aws/resource_aws_glue_dev_endpoint_test.go

```diff
@@ -0,0 +1,945 @@
+			}
+		}
+		return !lastPage
+	})
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping Glue Dev Endpoint sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("error retrieving Glue Dev Endpoint: %s", err)
+	}
+
+	return nil
+}
+
+func TestAccGlueDevEndpoint_Basic(t *testing.T) {
+	var endpoint glue.DevEndpoint
+
+	rName := acctest.RandomWithPrefix(GlueDevEndpointResourcePrefix)
+	resourceName := "aws_glue_dev_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_Basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
+					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "glue", fmt.Sprintf("devEndpoint/%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					testAccMatchResourceAttrGlobalARN(resourceName, "role_arn", "iam", regexp.MustCompile(`role/AWSGlueServiceRole-tf-acc-test-[0-9]+$`)),
+					resource.TestCheckResourceAttr(resourceName, "status", "READY"),
+					resource.TestCheckResourceAttr(resourceName, "arguments.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "number_of_nodes", "5"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccGlueDevEndpoint_Arguments(t *testing.T) {
+	var endpoint glue.DevEndpoint
+
+	rName := acctest.RandomWithPrefix(GlueDevEndpointResourcePrefix)
+	resourceName := "aws_glue_dev_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_Arguments2(rName, "bar", "python"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
+					resource.TestCheckResourceAttr(resourceName, "arguments.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "arguments.--foo", "bar"),
+					resource.TestCheckResourceAttr(resourceName, "arguments.--job-language", "python"),
+				),
+			},
+			{
+				Config: testAccGlueDevEndpointConfig_Arguments(rName, "baz"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
+					resource.TestCheckResourceAttr(resourceName, "arguments.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "arguments.--foo", "baz"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
```

Comment:
lets add another test case where it adds another argument.

Reply:
Good idea. I also cleaned up the test to have a similar structure as `TestAccAWSGlueJob_Tags`.